File: fe-ta/src/models/ProductSpec.ts
export interface ProductSpecRequest {
  specKey: string;
  specValue?: string;
  orderIndex?: number;
  note?: string;
}

export interface ProductSpecFilterRequest {
  productId?: number;
  productName?: string;

  specKey?: string;
  specValue?: string;
  orderIndex?: number;

  createdName?: string;
  updatedName?: string;

  fromUpdateTime?: string; // ISO string
  toUpdateTime?: string; // ISO string
  note?: string;
}

export interface ProductSpecResponse {
  id: number;
  productId: number;
  productName?: string;
  specKey: string;
  specValue?: string;
  orderIndex: number;
  createUid?: number;
  writeIUid?: number;
  updateTime?: string;
  createdName?: string;
  updatedName?: string;
  note?: string;
  option1?: string;
  option2?: string;
  option3?: string;
  option4?: string;
  option5?: string;
}

File: beSQLSugar/beSQLSugar/Enities/ProductSpec.cs
using SqlSugar;
namespace beSQLSugar.Models
{
    [SugarTable("product_spec")]
    public class ProductSpec
    {
        [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
        public int Id { get; set; }

        public int ProductId { get; set; }

        [SugarColumn(Length = 200, IsNullable = false)]
        public string SpecKey { get; set; } = string.Empty;

        [SugarColumn(Length = 1000)]
        public string? SpecValue { get; set; }

        [SugarColumn(DefaultValue = "0")]
        public int OrderIndex { get; set; } = 0;

        [SugarColumn(IsNullable = true)]
        public int? CreateUid { get; set; }

        [SugarColumn(IsNullable = true)]
        public int? WriteIUid { get; set; }

        [SugarColumn(IsNullable = true)]
        public DateTime? UpdateTime { get; set; }

        [SugarColumn(Length = 250, IsNullable = true)]
        public string? Note { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option1 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option2 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option3 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option4 { get; set; }

        [SugarColumn(Length = 120, IsNullable = true)]
        public string? Option5 { get; set; }
    }
}
